chia/daemon/server.py

```diff
@@ -699,6 +702,21 @@ async def register_service(self, websocket: WebSocketServerProtocol, request: Di
         log.info(f"{response}")
         return response
 
+    async def unregister_service(self, websocket: WebSocketServerProtocol, request: Dict[str, Any]) -> Dict[str, Any]:
+        self.log.info(f"Unregister service {request}")
+        service = request["service"]
+
+        num_connections = len(self.connections.get(service, []))
+        if service in self.connections:
```

Comment:
For explanation of the comment before:
If `service` is not in `self.connections`, then the `get` call will return `[]`, so that `num_connections` equals `0`.
That's why you can use `num_connections` directly here in the if-clause.

Minimal example:
```python
for n in range(20):
    if n % 3:
        print(n)
    else:
        print("skip")
```

Reply:
In fact this raises the question whether of not the key `service` should be deleted from `self.connections` if that list ever becomes empty.

Reply:
Hi @Nikolaj-K, thanks for reviewing.

What you want to tell is to change code like
```python
num_connections = len(self.connections.get(service, []))
if num_connections:
```
rather than
```python
num_connections = len(self.connections.get(service, []))
if service in self.connections:
```
do you?

Why I adopted latter one is for code completion freindliness.

```python
if service in self.connections:
    # IDE safely assume there is definitely `service` in `self.connections`
    ....

num_connections = len(self.connections.get(service, []))
if num_connections:
    # IDE cannot determine whethere `self.connections` has `service`
    # unless the IDE has ultra powerfull type inference system implemented.
```

Reply:
I don't know what it has to do with the IDE or type inferences - the two options do the exact same thing, they are equivalent.

Reply:
Example
```python
x = {"a" : [1,2,3]}

if "a" in x:
    print("will be printed")

if "c" in x:
    print("will NOT be printed")

if len(x.get("a", [])):
    print("will be printed")

if len(x.get("c", [])):
    print("will NOT be printed")
```

Reply:
Yes, type inference or code completion does not affect code execution itself.
It Just makes sense for developer experience, which usually leads to fewer bugs in software.

Reply:
Are you just suggesting what is truthy value and falsy value in python?

Reply:
I don't know how to read the last question, this is just pointing out that you can remove 2 redundant membership checks and gets. This code currently uses 4 but you only need 2. 
Just a hint though, this is my last thought on the topic :)

Reply:
By the way, thanks for reviewing my code. You just make me feel I'm not alone in this PR. :thumbsup: